src/Features/Core/Portable/AddParameter/RegisterFixData.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.AddParameter
+{
+    internal class RegisterFixData<TArgumentSyntax>
+        where TArgumentSyntax : SyntaxNode
+    {
+        public RegisterFixData() : this(new SeparatedSyntaxList<TArgumentSyntax>(), ImmutableArray<IMethodSymbol>.Empty)
+        {
+        }
```

Comment:
is this constructor used?  Seems strange, esp given this is a class, and we can just return null.

Reply:
It is used to mimic the former behavior of RegisterCodeFix. Returning null means try the next parent node. Returning data means we are done. There are code path in the constructor case (e.g. anonymous types) where we found a node but can't do anything. These are the cases where this constructor is used to return empty data.